backtest/engine.py

```
@@ -0,0 +1,427 @@
+            continue
+
+        # --- FILL DETECTION (Plan fill_model: tape conservative default) ---
+        # Tape-confirmed: a real trade printed at our resting price.
+        # Book-only: best_ask <= resting_price means book crossed us.
+        for trade in s.get("tape_delta") or []:
+            tprice = float(trade.get("price", 0))
+            tasset = str(trade.get("asset", ""))
+            up_token = first.get("up_token") or (ub.get("token_id") or "")
+            dn_token = first.get("down_token") or (db.get("token_id") or "")
+            if abs(tprice - resting_up) <= params.tick_size and tasset == up_token:
+                if params.fill_model in ("tape", "both"):
```

Ruff (0.16.2)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)